### PR TITLE
Add dart analysis backend

### LINKED ISSUE #840: Consider using dart_analysis_server instead of dart_language_server
`dart-sdk` now supports language server protocol by `dart_analysis_server`
https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/tool/lsp_spec/README.md

By using this implementation, we can use lsp-dart without any third party plugins.
And maybe `dart_analysis_server` is faster than `dart_language_server` (I compared with `vim-lsc` with `dart_analysis_server`)

### PR DESCRIPTION
**Pull request by itome**
Dart sdk now officially supports language server protocol.
https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/tool/lsp_spec/README.md

This pull request change backend for lsp-dart from `dart_language_server` to `analysis_server`.

close #840

### CODE REVIEW

File: lsp-clients.el [hunk 1]
```diff
@@ -305,37 +305,29 @@ finding the executable with `exec-path'."
...
-    (if pub
-        (if (executable-find dls)
-            dls
-          (message "Installing dart_language_server...")
-          (shell-command (concat pub " global activate dart_language_server"))
-          (message "Installed dart_language_server")
-          dls)
-      (error "Please ensure /path/to/dart-sdk/bin is on system path"))))
+  `("dart"
+    ,(expand-file-name (concat lsp-dart-sdk-dir "bin/snapshots/analysis_server.dart.snapshot"))
```
Comment (yyoncho) [inline lsp-clients.el]: use f-join instead of concat and fix the comment above(no slash will be required after this change.

**Comment by itome** [comment]
I added lsp-dart-analysis.el.
Should I add `(require 'lsp-dart-analysis)` to `lsp-clients.el` ?

**Comment by itome** [comment]
```
LSP :: dart_analysis_server has exited (finished)
Server dart_analysis_server:7697 exited with status exit. Do you want to restart it? (y or n) n
```
Language server exit when I type character, how to debug this? (I want to see log of dart_analysis_server or json they communicated).

**Comment by yyoncho** [comment]
> I added lsp-dart-analysis.el.
> Should I add `(require 'lsp-dart-analysis)` to `lsp-clients.el` ?

Yes. 

> ```
> LSP :: dart_analysis_server has exited (finished)
> Server dart_analysis_server:7697 exited with status exit. Do you want to restart it? (y or n) n
> ```
> 
> Language server exit when I type character, how to debug this? (I want to see log of dart_analysis_server or json they communicated).

Set lsp-print-io to t. You may check the stderr buffer as well.

File: lsp-dart.el [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+ (make-lsp-client :new-connection
+                  (lsp-stdio-connection
+                   'lsp-dart--analysis-server-command)
+                  :major-modes '(dart-mode)
+                  :priority -1
+                  :server-id 'dart_analysis_server))
+
+(provide 'lsp-dart)
+
+;;; lsp-dart-analysis.el ends here
```
Comment (yyoncho) [inline lsp-dart.el]: ditto

File: lsp-dart.el [hunk 3]
```diff
@@ -0,0 +1,91 @@
+;;; lsp-dart-analysis.el --- LSP mode -*- lexical-binding: t; -*-
```
Comment (yyoncho) [inline lsp-dart.el]: the file is named lsp-dart.el not lsp-dart-analysis.el

**Comment by yyoncho** [comment]
Thank you and congrats for your first `lsp-mode` PR! :tada:

**Pull request merged**